### DISCUSSION TITLE
Change domain for Community Container (Jellyfin)

### DISCUSSION THREAD

**Discussion by addy419** [nextcloud/all-in-one#5679] [Questions] [answered]
Hi @szaimen, I have my Nextcloud instance running on `cloud.mydomain.com`. According to the documentation, if I enable the jellyfin server with caddy, it will automatically try to reverse proxy it on `media.$NCDOMAIN` which is `media.cloud.mydomain.com`. Is it possible to instead have the media server on `media.mydomain.com`? What changes would I need to make to achieve this?

**Comment by szaimen** [comment, marked as answer]
Hi, there are multiple ways to achieve this:
- change your NC_DOMAIN to mydomain.com
- use your own reverse proxy and proxy media.mydomain.com manually to port 8096 of the server
- use https://github.com/nextcloud/all-in-one/tree/main/community-containers/caddy and use this feature to add your custom config: 
![image](https://github.com/user-attachments/assets/353c2490-14d0-4f71-817e-723013d18166)

**Reply by Markezzz** [reply]
I have the same issue like @addy419.
So I thought I could solve it using the method you suggested.

> Hi, there are multiple ways to achieve this:
> 
> * use https://github.com/nextcloud/all-in-one/tree/main/community-containers/caddy and use this feature to add your custom config:
>   ![image](https://private-user-images.githubusercontent.com/42591237/391507953-353c2490-14d0-4f71-817e-723013d18166.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODE4MTMxNTcsIm5iZiI6MTc4MTgxMjg1NywicGF0aCI6Ii80MjU5MTIzNy8zOTE1MDc5NTMtMzUzYzI0OTAtMTRkMC00ZjcxLTgxN2UtNzIzMDEzZDE4MTY2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA2MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwNjE4VDIwMDA1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE0YjYzNTQxYWFmN2VjODQ1OTdhYmQzYWZiZDYxMWM4ZDBkZmU1ZGI5MWZlOThkZThjMjA5MzQ1ZjgxZDg0M2QmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JnJlc3BvbnNlLWNvbnRlbnQtdHlwZT1pbWFnZSUyRnBuZyJ9.Vj7673gEzycGetZWcY1juPrRnMN7vWsq6k_JCCHoL8U)

But now i get this error:
`Error: adapting config using caddyfile: server block without any key is global configuration, and if used, it must be first`

This is my CaddyFile:
`{
    email admin@mydomain.de
}

# ---------------------------------------------------------
# Vaultwarden – bw.mydomain.de
# ---------------------------------------------------------
bw.mydomain.de {
    reverse_proxy 127.0.0.1:8812
}

# ---------------------------------------------------------
# Stalwart Mailserver – mail.mydomain.de
# Web-UI + Submission/IMAP/SMTP via TLS
# ---------------------------------------------------------
mail.mydomain.de {
    reverse_proxy 127.0.0.1:10003
}

# ---------------------------------------------------------
# Main domain – mydomain.de (optional)
# ---------------------------------------------------------
mydomain.de, www.mydomain.de {
    respond "Welcome to mydomain.de"
`

And if i use this code in my CaddyFile:
`# ---------------------------------------------------------
# Vaultwarden – bw.mydomain.de
# ---------------------------------------------------------
bw.mydomain.de {
    reverse_proxy 127.0.0.1:8812
}

# ---------------------------------------------------------
# Stalwart Mailserver – mail.mydomain.de
# Web-UI + Submission/IMAP/SMTP via TLS
# ---------------------------------------------------------
mail.mydomain.de {
    reverse_proxy 127.0.0.1:10003
}

# ---------------------------------------------------------
# Main domain – mydomain.de (optional)
# ---------------------------------------------------------
mydomain.de, www.mydomain.de {
    respond "Welcome to mydomain.de"
`

Or can @addy419  help me?

**Comment by addy419** [comment]
Thanks for the answer @szaimen . I managed to add a new import with the instructions provided. Is it possible to remove the `https://media.{$NC_DOMAIN}:443` from Caddyfile (stop auto configuration)

**Reply by szaimen** [reply]
> Is it possible to remove the `https://media.{$NC_DOMAIN}:443` from Caddyfile (stop auto configuration)

It isnt

**Reply by addy419** [reply]
Alright, I can live with that I guess. Thanks
